shared/util/ix.ts

```diff
@@ -113,3 +114,56 @@ export function asyncGeneratorFromPromise<P extends unknown[], R>(
         yield await func(...args)
     }
 }
+
+/** The maximum number of promise results to cache. */
+export const PROMISE_CACHE_CAPACITY = 5
```

Comment:
question:
any context for why this is 5?

Reply:
It's mainly to cache back-to-back requests (we want a search definition like 2-3 times per hover and we don't want to recalculate two of those three). Five is just a bit of headroom.